### PR TITLE
COM-357: Add support for additional content in DAM grid rows

### PR DESCRIPTION
**Pull request by johnnyomair**
We need this for the "External DAM systems" feature to show the import source in the grid.

---

DAM grid row additional content

### CODE REVIEW

File: packages/admin/cms-admin/src/dam/DataGrid/label/DamItemLabel.tsx [hunk 1]
```diff
@@ -29,7 +31,7 @@ const Path = styled(Typography)`
 `;
 
 interface DamItemLabelProps {
-    asset: GQLDamFolderTableFragment | GQLDamFileTableFragment;
+    asset: DamRow;
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/dam/DataGrid/label/DamItemLabel.tsx:34]: I don't think it's a good idea that all components in the grid use the fragment from the parent, but fixing that is out of scope...

File: packages/admin/cms-admin/src/dam/config/DamConfigContext.tsx [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 
+import { DamRow } from "../DataGrid/FolderDataGrid";
+
 export interface DamConfig {
     additionalMimeTypes?: string[];
     scopeParts?: string[];
     enableLicenseFeature?: boolean;
     requireLicense?: boolean;
     additionalToolbarItems?: React.ReactNode;
+    renderAdditionalRowContent?: (row: DamRow) => React.ReactNode;
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/dam/config/DamConfigContext.tsx:11]: I decided to add the `render` prefix because this is a render prop. However, it's kinda odd that `additionalToolbarItems` isn't a render prop. What do you think?

File: packages/admin/cms-admin/src/dam/DataGrid/label/DamItemLabel.tsx (hunk 1 above)
Comment (nsams) [reply packages/admin/cms-admin/src/dam/DataGrid/label/DamItemLabel.tsx:34]: true

File: packages/admin/cms-admin/src/dam/config/DamConfigContext.tsx (hunk 2 above)
Comment (nsams) [reply packages/admin/cms-admin/src/dam/config/DamConfigContext.tsx:11]: yes, they could both use the same technique

File: packages/admin/cms-admin/src/index.ts [hunk 3]
```diff
@@ -51,6 +51,7 @@ export { useDamConfig } from "./dam/config/useDamConfig";
 export { useCurrentDamFolder } from "./dam/CurrentDamFolderProvider";
 export { DamPage } from "./dam/DamPage";
 export { useDamFileUpload } from "./dam/DataGrid/fileUpload/useDamFileUpload";
+export type { DamRow } from "./dam/DataGrid/FolderDataGrid";
```
Comment (nsams) [inline packages/admin/cms-admin/src/index.ts:54]: imho that name is too generic (for a public export)

DamRowFragment? DamRowData?

**Comment by nsams** [comment]
Did you also consider additional columns?

File: packages/admin/cms-admin/src/index.ts (hunk 3 above)
Comment (johnnyomair) [reply packages/admin/cms-admin/src/index.ts:54]: MAybe DamGridItem or DamGridRow?

Comment (nsams) [reply packages/admin/cms-admin/src/index.ts:54]: it's data structure (that contains just the data) vs. component (that renders it)

usually our exports are components, so I could think DamGridRow is also a component

File: packages/admin/cms-admin/src/dam/config/DamConfigContext.tsx (hunk 2 above)
Comment (nsams) [reply packages/admin/cms-admin/src/dam/config/DamConfigContext.tsx:11]: what is the difference between the first two?

File: packages/admin/cms-admin/src/index.ts (hunk 3 above)
Comment (johnnyomair) [reply packages/admin/cms-admin/src/index.ts:54]: I could simply re-export the generated GraphQL type without a renaming, what do you think?

**Comment by johnnyomair** [comment]
> well, the resulting UI should usually not depend on how complex it was to implement :)

You're right, I've asked UX about it and will update the PR depending on the outcome.

File: packages/admin/cms-admin/src/index.ts (hunk 3 above)
Comment (nsams) [reply packages/admin/cms-admin/src/index.ts:54]: The reason why you add it to the public api is, because it gets passed in renderAdditionalRowContent render prop.

final-form names the arguments passed to a render function eg `FieldRenderProps`, maybe you could use `AdditionalRowContentRenderProps` accordingly?

**Comment by johnnyomair** [comment]
Superseded by #1756.
